File: packages/react-native-vision-camera/ios/Hybrid Objects/HybridNativeThreadFactory.swift
///
/// HybridNativeThreadFactory.swift
/// VisionCamera
/// Copyright © 2025 Marc Rousavy @ Margelo
///

import Foundation
import NitroModules

final class HybridNativeThreadFactory: HybridNativeThreadFactorySpec {
  func createNativeThread(name: String) throws -> any HybridNativeThreadSpec {
    let queue = DispatchQueue(label: name)
    return HybridNativeThread(queue: queue)
  }
}

File: packages/react-native-vision-camera/android/src/main/java/com/margelo/nitro/camera/HybridNativeThreadFactory.kt
package com.margelo.nitro.camera

import androidx.annotation.Keep
import com.facebook.proguard.annotations.DoNotStrip
import com.margelo.nitro.camera.hybrids.HybridNativeThread
import com.margelo.nitro.camera.utils.IdentifiableExecutor

@Keep
@DoNotStrip
class HybridNativeThreadFactory : HybridNativeThreadFactorySpec() {
  override fun createNativeThread(name: String): HybridNativeThreadSpec {
    val executor = IdentifiableExecutor(name)
    return HybridNativeThread(executor)
  }
}
